hw09-my_orm/src/main/java/ru/otus/jdbc/templates/JdbcTemplateImpl.java

```diff
@@ -0,0 +1,203 @@
+            List<Object> params = mapper.getValuesForInsert(objectData);
+            logger.info("Received values("+params.size()+") for insert");
+            Savepoint savepoint = connection.setSavepoint("InsertData");
+            try {
+                long id = executer.insert(query, params);
+                logger.info("Inserted record, id=" + id);
+                connection.commit();
+                logger.info("Committed changes");
+                ReflectionClass.setFieldValue(objectData, mapper.getFieldAutoincrement(), id);
+                logger.info("Seted value to Id field");
+            } catch (SQLException e) {
+                logger.info("Rollback to savepoint. Error: "+e.getMessage());
+                connection.rollback(savepoint);
+            }
+
+        }
+    }
+
+    @Override
+    public void update(T objectData) throws SQLException {
+        logger.info("Procedure for update record");
+        try(Connection connection = dataSource.getConnection()){
+            executer = new DBExecutorImpl<>(connection);
+            String query = mapper.getUpdateScript();
+            logger.info("Received query script for update record");
+            List<Object> params = mapper.getValuesForInsert(objectData);
+            logger.info("Received values("+params.size()+") for update");
+            Savepoint savepoint = connection.setSavepoint("UpdateData");
+            try {
+                Object objectValue = ReflectionClass.getFieldValue(objectData, mapper.getFieldAutoincrement());
+                long id = objectValue != null ? (long) objectValue : -1;
+                if( id == -1 ) {
+                    throw new EntetyException("Id is null");
+                } else {
+                    executer.update(query, params, id);
+                    logger.info("Update record, id=" + id);
+                    connection.commit();
+                    logger.info("Committed changes");
+                }
+            } catch (SQLException | EntetyException e) {
+                logger.info("Rollback to savepoint. Error: "+e.getMessage());
+                connection.rollback(savepoint);
+            }
+
+        }
+    }
+
+    @Override
+    public void createOrUpdate(T objectData) throws SQLException {
+        logger.info("Procedure for create or update record");
+        try(Connection connection = dataSource.getConnection()){
+            executer = new DBExecutorImpl<>(connection);
+            logger.info("Received query script for update record");
+            List<Object> params = mapper.getValuesForInsert(objectData);
+            logger.info("Received values("+params.size()+") for update");
+            Savepoint savepoint = connection.setSavepoint("CreateOrUpdateData");
+            try {
+                Object objectValue = ReflectionClass.getFieldValue(objectData, mapper.getFieldAutoincrement());
+                long id = objectValue != null ? (long) objectValue : 0;
+                if( id == 0 ) {
```

Comment:
```suggestion
                if( id == ID_UNSAVED ) {
```

Вынести в именованную константу